### PR TITLE
search-feature-fixes

### PR DESCRIPTION
**Pull request by yellowHatpro**
- Added a toast to check the internet of searches without the internet being on.
- Added loading and result not found animation in search feature.

### CODE REVIEW

File: app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt [hunk 1]
```diff
@@ -164,14 +170,24 @@ class SearchActivity : AppCompatActivity(), SearchView.OnQueryTextListener {
     }
 
     override fun onQueryTextSubmit(query: String): Boolean {
+        if (!App.context!!.isOnline){
```
Comment (akshaaatt) [inline app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt:173]: We need to stop force-unwrapping things. 
Try something like `App.context?.isOnline == false` over here.

File: app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt [hunk 2]
```diff
@@ -164,14 +170,24 @@ class SearchActivity : AppCompatActivity(), SearchView.OnQueryTextListener {
...
         binding.recyclerView.adapter = adapter
         adapter!!.resetAnimation()
-        viewModel!!.search(searchTypeFromSpinner, query).observe(this, { pagingData: PagingData<ResultItem> ->
+        viewModel!!.search(searchTypeFromSpinner, query).observe(this) { pagingData: PagingData<ResultItem> ->
             adapter!!.submitData(lifecycle, pagingData)
-        })
+        }
+        lifecycleScope.launch {
+            adapter!!.loadStateFlow.collect {
+                binding.loadingAnimation.root.isVisible = it.refresh is LoadState.Loading
```
Comment (akshaaatt) [inline app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt:187]: What after this? What if the user wants to search for something else? This view hides the entire screen to search. I suggest we add the loading view only along with the size of the recycerView. Another solution could be to undo this view upon pressing the back button.

**Review by akshaaatt — CHANGES_REQUESTED** [review]
Great work! Just a couple of things and then we should be good to go.

File: app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt (hunk 2 above)
Comment (akshaaatt) [reply app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt:187]: I meant to tag the noResult line which is one line below this!

File: app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt (hunk 1 above)
Comment (yellowHatpro) [reply app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt]: Changed the if condition to ```App.context?.isOnline == false```.

File: app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt (hunk 2 above)
Comment (yellowHatpro) [reply app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt]: Sorry, I actually started working on it asap so didn't see the second reply.
I actually have now overridden the onBackPressed function, so the user can navigate back to the initially looking fragment,
Also, I made changes in the XML of the loading and noresult animations and set them in a way that they remain below the search bar. 
Please suggest if I can change this approach.

File: app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt [hunk 3]
```diff
@@ -210,5 +210,16 @@ class SearchActivity : AppCompatActivity(), SearchView.OnQueryTextListener {
         }
         return false
     }
+
+    override fun onBackPressed() {
+        if (!binding.gridView.isVisible){
```
Comment (akshaaatt) [inline app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt:215]: After considering the user experience, I found that when successful search results are shown to a user, the back press would have them show the screen to add artists, etc., hence introducing an extra back press for someone who's looking to exit the search screen. It's better to have the back press overridden for the case only when the search fails.

Comment (yellowHatpro) [reply app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt]: Sir I changed the condition from ```!binding.gridView.isVisible``` to ```binding.noResult.isVisible```.
Is it fine or do I need to make other changes ??

**Review by akshaaatt — APPROVED** [review]
Great work. Thanks!

**Pull request merged**